src/server-daemon.ts

```diff
@@ -48,31 +57,67 @@ export async function startDaemon(context: vscode.ExtensionContext, lspExec: str
   features.setupAyaSpecialFeatures(context, languageClient);
 }
 
-function runDebug(outputChannel: vscode.OutputChannel, lspExec: string): ServerOptions {
+function runDebugFatJar(outputChannel: vscode.OutputChannel, lspLoadPath: string, javaPath: string): ServerOptions {
+  return () => new Promise((resolve, reject) => {
+    const proc = spawnFatJar(outputChannel, lspLoadPath, ["--enable-preview", "-jar", "--mode", "debug"], javaPath);
+    proc.on("exit", (code, sig) => outputChannel.appendLine(`The language server exited with ${code} (${sig})`));
+    proc.on("error", reject);
+    proc.on("spawn", () => resolve(proc));
+  });
+}
+
+function runDebug(outputChannel: vscode.OutputChannel, lspLoadPath: string): ServerOptions {
   return () => new Promise((resolve, reject) => {
-    const proc = spawnJava(outputChannel, lspExec, ["--mode", "debug"]);
+    const proc = spawnExec(outputChannel, lspLoadPath, ["--mode", "debug"]);
     proc.on("exit", (code, sig) => outputChannel.appendLine(`The language server exited with ${code} (${sig})`));
     proc.on("error", reject);
     proc.on("spawn", () => resolve(proc));
   });
 }
 
-function runServer(outputChannel: vscode.OutputChannel, lspExec: string, host: string, port: number): ServerOptions {
+function runServerFatJar(outputChannel: vscode.OutputChannel, lspLoadPath: string, host: string, port: number, javaPath: string): ServerOptions {
+  return () => new Promise((resolve, reject) => {
+    const server = net.createServer(socket => {
+      server.close();
+      resolve({ reader: socket, writer: socket });
+    });
+    server.listen(port, host, () => {
+      const tcpPort = (server.address() as net.AddressInfo).port.toString();
+      spawnFatJar(outputChannel, lspLoadPath, ["--enable-preview", "-jar", "--mode", "client", "--port", tcpPort], javaPath);
+    });
+    server.on("error", reject);
+  });
+}
+
+function runServer(outputChannel: vscode.OutputChannel, lspLoadPath: string, host: string, port: number): ServerOptions {
   return () => new Promise((resolve, reject) => {
     const server = net.createServer(socket => {
       server.close();
       resolve({ reader: socket, writer: socket });
     });
     server.listen(port, host, () => {
       const tcpPort = (server.address() as net.AddressInfo).port.toString();
-      spawnJava(outputChannel, lspExec, ["--mode", "client", "--port", tcpPort]);
+      spawnExec(outputChannel, lspLoadPath, ["--mode", "client", "--port", tcpPort]);
     });
     server.on("error", reject);
   });
 }
 
-function spawnJava(outputChannel: vscode.OutputChannel, lspExec: string, args: string[]): child_process.ChildProcess {
-  const proc = child_process.spawn(lspExec, args);
+function spawnFatJar(outputChannel: vscode.OutputChannel, lspLoadPath: string, args: string[], javaPath: string): child_process.ChildProcess {
```

Comment:
The spawnFatJar has lots of duplicate lines with spawnExec. Could you please reuse some code?